packages/x-components/src/x-modules/history-queries/components/toggle-history-queries.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <BaseSwitch @change="toggle" :value="isEnabled" />
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component } from 'vue-property-decorator';
+  import BaseSwitch from '../../../components/base-switch.vue';
+  import { State, xComponentMixin } from '../../../components';
+  import { historyQueriesXModule } from '../x-module';
+
+  /**
+   * Toggle history queries component to enable or disable the history queries.
+   * This component emits events depending on the `isEnabled` value.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(historyQueriesXModule)],
+    components: { BaseSwitch }
+  })
+  export default class ToggleHistoryQueries extends Vue {
+    /**
+     * A boolean with the isEnabled value coming from the store state.
+     *
+     * @public
+     */
+    @State('historyQueries', 'isEnabled')
+    public isEnabled!: boolean;
+
+    /**
+     * Emit events depending on the switch value.
+     *
+     * @internal
+     */
+    protected toggle(): void {
+      if (this.isEnabled) {
```

Comment:
I would always emit the `UserToggledHistoryQueries` event, which I proposed to rename to `UserClickedToggleHistoryQueries` to better reflect the intention.

Then we can modify the wiring using a `filter` function so it only listens to the events with payload `true`, and maybe we can get rid of the trick of `UserDisableHistoryQueries` event. But I'm not sure about what we talked the other day in the refinement. I thought we wrote the new proposal in the task but I couldn't find it.

Reply:
I agree with the renaming of the event.

I've also refactored the wiring to use a `filter` function as you suggested. I had to change the `toggleHistoryQueries` action param type so TS doesn't shout.

Reply:
Thanks for pointing this because I totally forgot about it haha. When I refined the task I think that [we also agreed to](https://searchbroker.atlassian.net/browse/EX-6514?focusedCommentId=153413) refactor the action so it does not depend on a payload. 

Also I just remembered why we decided to emit multiple events reading that comment. It was to ease the usage of that modal component. And right now seeing your updated example with this use case It completely makes sense even if it is not the desired approach.

So I have to say sorry for the misunderstanding and ask you to add the 2 events again.
* Add `UserClickedEnableHistoryQueries` and `UserClickedDisableHistoryQueries`.
* Remove `UserClickedToggleHistoryQueries` event.
* Rename `toggleHistoryQueries` to `setHistoryQueriesEnabled`

Do you think that would work?